### ISSUE TITLE
Google Sync stopped working

### ISSUE THREAD

**Issue by Dragonslayer42069** [alextselegidis/easyappointments#1113]
Hello,

yesterday in the evening everything worked fine. But today the syncing stopped working.

`Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.", "errors": [ { "message": "Login Required.", "domain": "global", "reason": "required", "location": "Authorization", "locationType": "header" } ], "status": "UNAUTHENTICATED"`

GOOGLE_PRODUCT_NAME, GOOGLE_CLIENT_ID,GOOGLE_CLIENT_SECRET ,GOOGLE_API_KEY is correct.

Anyone else experiencing this issue?

Cheers

**Comment by mirlipili** [comment]
Hello, 
I have a similar problem, google sync stopped working a few days ago, but without an error.
I have a cronjob running to sync it and it still reports "SUCCESS"
I have disabled and enabled synchronisation, had to connect to my google account and select a calendar, but when I try to synchronise, I get this error:
`{ "error": { "errors": [ { "domain": "usageLimits", "reason": "quotaExceeded", "message": "Calendar usage limits exceeded." } ], "code": 403, "message": "Calendar usage limits exceeded." } } `

I made a new google account, made all the necessary steps, but I get the same error with a new user and a new calendar, so it shouldn't be a quota problem.

Any idea?

**Comment by mirlipili** [comment]
I dropped the database and started with a new empty easyappointment calendar, and now it works, so it probably was a quota problem.

I was expecting a `403: User Rate Limit Exceeded` or `403: Rate Limit Exceeded` error, and I had no indication in google cloud console that I was over my quota.

Is there a way to delete old entries in EA? Maybe a way to archive previous years so it doesn't cause a problem when you suddenly have to synchronize multiple years in one go, or is there a way for me to delete old appointments out of the database, without losing all my customers?

Thanks

**Comment by alextselegidis** [comment]
Hello! 

There is no archive or remove-data-before-date function but that might probably have been a good idea at some point in the future. 

Regarding syncing the default behavior is set to be 30 days in the past and 90 days in the future, so if you have too many appointments then you could change that in the `ea_users_settings` table for each user individually.

**Comment by mirlipili** [comment]
I just checked, and I probably had more than 600 appointments in this time period (due to accidentally doubling all my appointments as I tried to re-enable syncing), so this shouldn't happen again.

In my case, the "Authorised JavaScript origins" and the "Authorised redirect URIs" fields where empty in google cloud console (since 2018), and it suddenly stopped working. As I tried re-enabling it, the quota problem stopped it again.
Waiting 24h for the quota to reset, and changing the syncing value should work if I ever have the same problem.

Thanks
